File: api/app/schemas/employee.py
from pydantic import BaseModel


class EmployeeOut(BaseModel):
    """Formato de funcionário devolvido pela API."""

    id: int
    name: str
    position_name: str


class TeamMemberOut(BaseModel):
    """Funcionário liderado, com a distância hierárquica até o líder consultado."""

    id: int
    name: str
    position_name: str
    depth: int
    parent_id: int


class TeamMemberEvaluationOut(BaseModel):
    """Membro do time com a avaliação vigente, quando houver."""

    id: int
    name: str
    position_name: str
    depth: int
    parent_id: int
    evaluation_id: int | None
    evaluation_leader_id: int | None
    evaluation_leader_name: str | None
    week_key: str | None
    weighted_score: float | None

File: web/src/types/employee.ts
export type Employee = {
  id: number
  name: string
  positionName: string
}

export type TeamMember = Employee & {
  depth: number
  parentId: number | null
}

export type TeamMemberEvaluation = TeamMember & {
  evaluationId: number | null
  evaluationLeaderId: number | null
  evaluationLeaderName: string | null
  weekKey: string | null
  weightedScore: number | null
}
